### PR TITLE
Add new-format combined tests and improve combined WASM and NullVM development.

### PR DESCRIPTION
**Pull request by jplevyak**

### CODE REVIEW

File: test/extensions/common/wasm/wasm_test.cc [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+  void scriptLog(spdlog::level::level_enum level, absl::string_view message) override {
+    std::cerr << std::string(message) << "\n";
+    scriptLog_(level, message);
+  }
+  MOCK_METHOD2(scriptLog_, void(spdlog::level::level_enum level, absl::string_view message));
+};
+
+class WasmCommonTest : public testing::TestWithParam<std::string> {};
+
+INSTANTIATE_TEST_SUITE_P(Runtimes, WasmCommonTest, testing::Values("v8", "null"));
```
Comment (PiotrSikora) [inline test/extensions/common/wasm/wasm_test.cc]: No WAVM?

File: test/extensions/common/wasm/wasm_test.cc [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+  wasm_handler.reset();
+  dispatcher->run(Event::Dispatcher::RunType::NonBlock);
+  // This will SEGV on nullptr if wasm has been deleted.
+  wasm_weak.lock()->configure(root_context, plugin, "done");
+  dispatcher->run(Event::Dispatcher::RunType::NonBlock);
+  dispatcher->clearDeferredDeleteList();
+}
+
+TEST_P(WasmCommonTest, BadSignature) {
+  if (GetParam() != "v8") {
```
Comment (PiotrSikora) [inline test/extensions/common/wasm/wasm_test.cc]: This should be:
```
if (GetParam() == "null") {
  return;
}
```
(same in other places)

Comment (jplevyak) [reply test/extensions/common/wasm/wasm_test.cc]: Done

File: test/extensions/common/wasm/wasm_test.cc (hunk 1 above)
Comment (jplevyak) [reply test/extensions/common/wasm/wasm_test.cc]: Done.

**Comment by PiotrSikora** [comment]
Could you merge with master? Maybe the new build image will help pass the CI.

**Comment by jplevyak** [comment]
It's passing now

**Pull request merged**